charts/sampling/templates/_otelcol_processor_transform.river.txt

```diff
@@ -0,0 +1,69 @@
+{{- define "deployment.processor.transform.add_resource_attributes" -}}
+otelcol.processor.transform "add_resource_attributes_as_metric_attributes" {
+  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.transform/
+  error_mode = "ignore"
+
+  metric_statements {
+    context    = "datapoint"
+    statements = [
+      "set(attributes[\"deployment.environment\"], resource.attributes[\"deployment.environment\"])",
+      "set(attributes[\"service.version\"], resource.attributes[\"service.version\"])",
+    ]
+  }
+
+  output {
+    metrics = [otelcol.processor.batch.default.input]
+  }
+}
+
+{{ end }}
+
+{{- define "deployment.processor.transform.add_k8s_attributes" -}}
+otelcol.processor.transform "add_k8s_attributes" {
+  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.transform/
+  // Grafana Cloud Kubernetes monitoring expects Loki labels `cluster`, `pod`, and `namespace`
+  error_mode = "ignore"
+
+  log_statements {
+    context    = "resource"
+    statements = [
+      "set(attributes[\"pod\"], attributes[\"k8s.pod.name\"])",
+      "set(attributes[\"namespace\"], attributes[\"k8s.namespace.name\"])",
+      "set(attributes[\"loki.resource.labels\"], \"pod, namespace\")",
+    ]
+  }
+
+  trace_statements {
+    context    = "resource"
+    statements = [
+      "set(attributes[\"k8s.cluster.name\"], \"{{ .Values.k8s.cluster.name }}\")",
+    ]
+  }
+
+  output {
+    logs   = [otelcol.processor.batch.default.input]
+    traces = [otelcol.exporter.loadbalancing.default.input]
```

Comment:
shall batch processor be used for traces pipeline as well?

Reply:
initially I didn't enable this as the batch processor docs indicate this should happen after sampling; however, it also says that it should be configured on every collector:
"It is highly recommended to configure the batch processor on every collector. The batch processor should be defined in the pipeline after the memory_limiter as well as any sampling processors. This is because batching should happen after any data drops such as sampling."

wdyt?

Reply:
I think we need to add the batch processor.
The part "The batch processor should be defined in the pipeline after the memory_limiter as well as any sampling processors. This is because batching should happen after any data drops such as sampling." is applicable within one collector. In our case, we have two separate collectors, and we treat them separately

Reply:
From what I remember, @gouthamve recently said about a Comply Advantage escalation that the memory limiter processor should be removed, that it may be deprecated.
@jpkrohling  @gouthamve  can you please help us?